### PR TITLE
Feature/all operations

### PR DESCRIPTION
**Pull request by sanchay-nyt**

### CODE REVIEW

**Comment by digitalbuddha** [comment]
hey hey, I think you had a bad rebase, I don't believe there should have been more than a dozen files warranting a change.

File: filesystem/src/main/java/com/nytimes/android/external/fs/FSReader.java [hunk 1]
```diff
@@ -51,4 +53,21 @@ public void call(Emitter<BufferedSource> emitter) {
...
+    public Observable<BufferedSource> readAll(@Nonnull final T type) throws FileNotFoundException {
+        return Observable
+                .from(fileSystem.list(pathResolver.resolve(type)))
+                .map(new Func1<String, BufferedSource>() {
+                    @Override
+                    public BufferedSource call(String s) {
+                        try {
+                            return fileSystem.read(s);
+                        } catch (FileNotFoundException e) {
+                            throw Exceptions.propagate(e);
```
Comment (digitalbuddha) [inline filesystem/src/main/java/com/nytimes/android/external/fs/FSReader.java:68]: this is my Today I learned, neat!

File: store/src/main/java/com/nytimes/android/external/store/base/DiskErase.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.nytimes.android.external.store.base;
+
+import javax.annotation.Nonnull;
+
+import rx.Observable;
+
+public interface DiskErase<Raw, Key> {
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/DiskErase.java:7]: Nice abstraction 💯

File: store/src/main/java/com/nytimes/android/external/store/util/NoopPersister.java [hunk 3]
```diff
@@ -35,4 +37,10 @@
     public void clear(@Nonnull Key key) {
         networkResponses.remove(key);
     }
+
+    @Nonnull
+    @Override
+    public Observable<Raw> readAll(@Nonnull Key key) throws FileNotFoundException {
+        return null;
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/util/NoopPersister.java:44]: I'm not a fan of returning null here especially when Rxjava2 does not allow you to propogate a null to a consumer.  Ideally we can structure the base classes in a manner that would not need this override. If that is too messy then throw an exception from here instead.

**Comment by digitalbuddha** [comment]
I think your rebase was off of rx2 branch and not develop.  I reviewed the actual meat of this pr and it looks good. The only thing that I think it is missing is a little better test coverage (I don't think there was a test for deleteAll).  Likewise there should be tests for the exception case as well.  Thank you for this PR it came together really nicely!

**Comment by digitalbuddha** [comment]
looking at your commits it seems like #130 should not have been in your branch as this was from rx2 branch and not develop.

**Comment by sanchay-nyt** [comment]
Thanks @digitalbuddha . With help from Brian, I have created a new PR https://github.com/NYTimes/Store/pull/207 . I will take care of the comments you posted today.

**Pull request closed without merging**